self-hosted.md

```diff
@@ -0,0 +1,135 @@
+# Creating a Self-Hosted Cluster
+
+A self-hosted cluster is a cluster that acts as both the management cluster and the workload cluster. A self-hosted cluster manages its own lifecycle.
+
+In this this section we will create a workload cluster and make it self-hosted by converting it into its own management cluster.
+
+<!-- table of contens generated via: https://github.com/thlorenz/doctoc -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Creating a Self-Hosted Cluster](#creating-a-self-hosted-cluster)
+  - [Create a Cluster](#create-a-cluster)
+  - [Convert Workload Cluster to Management Cluster](#convert-workload-cluster-to-management-cluster)
+  - [Clean up](#clean-up)
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Create a Cluster
+
+Firstly, lets create a workload cluster called `docker-cluster-self-hosted`.
+
+These steps will similar to how we created `docker-cluster-one`.
+
+* Create the cluster
+
+```bash
+kubectl apply -f yamls/clusters/docker-cluster-self-hosted.yaml
+```
+
+* Retrieve the cluster's kubeconfig
+
+```bash
+kind get kubeconfig --name docker-cluster-self-hosted > self-hosted.kubeconfig
+```
+
+* Install Calico on the Cluster using our prepared yaml spec:
+
+```bash
+    kubectl --kubeconfig self-hosted.kubeconfig apply -f yamls/cni/calico.yaml
+```
+
+## Convert Workload Cluster to Management Cluster
+
+Now that we have the `docker-cluster-self-hosted` workload cluster lets install Cluster API components on it to make it a management cluster.
+
+Firstly lets load the required images into the cluster:
+
+For Linux and mac:
+
+```bash
+sh ./scripts/load-images.sh
+```
+
+For windows:
+
+```bash
+.\scripts\load-images.ps1
+```
```

Comment:
Let's try to replace this with preload images. I think we can just add all needed images into our ClusterClass. We already have some so that everything works offline. Let's just add the ones we need here as well

(please add them to both ClusterClasses in the repo)

Reply:
both? I only see one clusterclass in the repo. https://github.com/ykakarap/kubecon-na-22-capi-lab/tree/main/yamls/clusterclasses

Reply:
The other one is in the clusterctl repo.

We are basically only using the one in yamls/. But I want to ensur that the local repo is functional + if a user somehow uses it it has the same / a working ClusterClass

Reply:
I push a commit